Source/WebCore/page/WebKitStringMatcher.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <span>
+#include <wtf/RefCounted.h>
+#include <wtf/text/Latin1Character.h>
+
+namespace JSC {
+class JSGlobalObject;
+class JSValue;
+}
+
+namespace WebCore {
+
+struct WebKitStringMatcherOptions;
+
+class WEBCORE_EXPORT WebKitStringMatcher : public RefCounted<WebKitStringMatcher> {
+public:
+    struct State {
+        // a transitionsEndIndexOrMatchSentinel of 0 means transitionsBeginIndexOrMatchIdentifier is an identifier
+        // and jump to the next State without progressing through the string.
+        static constexpr uint16_t matchSentinel { 0 };
```

Comment:
Hm... can we use `Markable` for this instead of manually managing a sentinel value?

Reply:
No.  In a format with raw uint16_t's from arbitrary data we need to use a sentinel value.